satpy/multiscene.py

```diff
@@ -490,8 +521,56 @@ def _simple_frame_compute(self, writers, frame_keys, frames_to_write):
                 w = writers[frame_key]
                 w.append_data(product_frame.compute())
 
+    def _get_writers_and_frames(
+            self, filename, datasets, fill_value, ignore_missing,
+            enh_args, imio_args):
+        """Get writers and frames.
+
+        Helper function for save_animation.
+        """
+        scene_gen = self._scene_gen
+        writers = {}
+        frames = {}
+
+        scene_gen = self._scene_gen
+        first_scene = self.first_scene
+        scenes = iter(self._scene_gen)
+        info_scenes = [first_scene]
+        if 'end_time' in filename:
+            # if we need the last scene to generate the filename
+            # then compute all the scenes so we can figure it out
+            log.debug("Generating scenes to compute end_time for filename")
```

Comment:
This sounds dangerous (wrt memory consumption). Can't the filename be determined later on? eg by first writing to a temp file, then renaming?

Reply:
This was originally added by me. @gerritholl is just moving the code to a separate method. Writing to a temp file and then moving it could be a solution but this was easier and less error prone, especially when you consider that this could be run on a cluster. The code also gets really ugly trying to format the filename later on and looking at Scenes that were previously generated. At least with how I see it being implemented.

Reply:
ok

Reply:
If I understand the code correctly it's not actually computing all the scenes, is it?  They remain daskified?

Reply:
The "generating" part of this statement @gerritholl is that it is evaluating the entire generator of Scenes. This can take a really long time. The arrays inside those Scenes will still be daskified, but all of the Scenes have to be created and all of the DataArrays inside those Scenes created.

Reply:
I don't know what a better solution is but I think it's somewhat out of scope for this PR, as it goes beyond simply refactoring.